cou/steps/plan.py

```diff
@@ -313,11 +313,24 @@ async def generate_plan(analysis_result: Analysis, args: CLIargs) -> UpgradePlan
     # NOTE (gabrielcocenza) upgrade group as None means that the user wants to upgrade
     #  the whole cloud.
     if args.upgrade_group in {CONTROL_PLANE, None}:
-        plan.sub_steps.extend(
+        plan.add_steps(
             _generate_control_plane_plan(target, analysis_result.apps_control_plane, args.force)
         )
+
+    hypervisor_apps, non_hypervisors_apps = _separate_hypervisors_apps(
+        analysis_result.apps_data_plane
+    )
     if args.upgrade_group in {DATA_PLANE, HYPERVISORS, None}:
-        plan.sub_steps.extend(await _generate_data_plane_plan(target, analysis_result, args))
+        plan.add_step(
+            await _generate_data_plane_hypervisors_plan(
+                args, analysis_result, target, hypervisor_apps
```

Comment:
Nit: Can we make the order of arguments consistent for all the `_generate_*_plan` functions?